perceval/components/abstract_processor.py

```diff
@@ -686,19 +686,24 @@ def with_input(self, input_state: BasicState) -> None:
         if self._min_detected_photons_filter is None:
             self._deduce_min_detected_photons(expected_photons)
 
-    def flatten(self) -> list[tuple]:
+    def flatten(self, max_iterations = None) -> list[tuple]:
         """
         :return: a component list where recursive circuits have been flattened
         """
-        return _flatten(self)
+        return _flatten(self, max_iterations = max_iterations)
 
 
-def _flatten(composite, starting_mode=0) -> list[tuple]:
+def _flatten(composite, starting_mode=0, max_iterations = None) -> list[tuple]:
```

Comment:
Actually, this method is tested nowhere, and by reading it, I can't figure out if it is working or not, can you check it ?